lib/src/utils/feature_flags_manager.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:instabug_flutter/src/generated/instabug.api.g.dart';
+import 'package:instabug_flutter/src/utils/ibg_build_info.dart';
+import 'package:meta/meta.dart';
+
+typedef OnW3CFeatureFlagChange = void Function(
+  bool isW3cExternalTraceIDEnabled,
+  bool isW3cExternalGeneratedHeaderEnabled,
+  bool isW3cCaughtHeaderEnabled,
+);
+
+class FeatureFlagsManager implements FeatureFlagsFlutterApi {
+  static InstabugHostApi _host = InstabugHostApi();
+  static FeatureFlagsManager _instance = FeatureFlagsManager();
+
+  /// @nodoc
+  @visibleForTesting
+  // ignore: use_setters_to_change_properties
+  static void $setHostApi(InstabugHostApi host) {
+    _host = host;
+  }
+
+  @visibleForTesting
+  // ignore: use_setters_to_change_properties
+  static void $setFeatureFlagsManager(FeatureFlagsManager featureFlagsManager) {
+    _instance = featureFlagsManager;
+  }
+
+  static bool _isAndroidW3CExternalTraceID = false;
+  static bool _isAndroidW3CExternalGeneratedHeader = false;
+  static bool _isAndroidW3CCaughtHeader = false;
+
+  static Future<bool> get isW3CExternalTraceID async {
+    if (IBGBuildInfo.instance.isAndroid) {
+      return Future.value(_isAndroidW3CExternalTraceID);
+    }
+    return ((await _host
+            .isW3CFeatureFlagsEnabled())['isW3cExternalTraceIDEnabled']) ??
+        false;
+  }
+
+  static Future<bool> get isW3CExternalGeneratedHeader async {
+    if (IBGBuildInfo.instance.isAndroid) {
+      return Future.value(_isAndroidW3CExternalGeneratedHeader);
+    }
+
+    return ((await _host.isW3CFeatureFlagsEnabled())[
+            'isW3cExternalGeneratedHeaderEnabled']) ??
+        false;
+  }
+
+  static Future<bool> get isW3CCaughtHeader async {
+    if (IBGBuildInfo.instance.isAndroid) {
+      return Future.value(_isAndroidW3CCaughtHeader);
+    }
+    return ((await _host
+            .isW3CFeatureFlagsEnabled())['isW3cCaughtHeaderEnabled']) ??
+        false;
+  }
```

Comment:
We usually access these three values in the same place so I feel like making 3 `_host.isW3CFeatureFlagsEnabled()` calls for each one is redundant and might hurt performance since it interrupts the user's network flow. I feel like we could group them under a single method that gets the value for the three flags and returns the result as an object.

We can optimize it even further on Android by making a call to the native method and caching the returned results in the `_isAndroidW3C*` flags to be reused in future calls. What do you think? (I see we are already doing something similar in `registerW3CFlagsListener` 👀)